Dispatcher/Controllers/FavoritesViewController.swift

```diff
@@ -1,46 +1,106 @@
-        ArticleModel(title: "Title Article 2", date: Date(), url: "Content of 2nd article", isFavorite: true, content: "http://noamkurtzer.co.il")
-    ]
+    var newsArray: [Articles] = []
+    
+    private var currentPaginationPage = 1
+    private var amountToFetch = 10
+    private var totalPaginationPages = 1
+    
+    var dataSource: TableViewDataSourceManager<Articles>!
     
     
     override func viewDidLoad() {
         super.viewDidLoad()
         
         customHeader.initView(delegate: self, icon1: UIImage(named: "notifications"), icon2: UIImage(named: "search"), leftIcon: UIImage(named: "logo"))
-        
-        tableView.rowHeight = 115.0
-        tableView.separatorStyle = .none
-        tableView.register(UINib(nibName: Constants.NibNames.favorites, bundle: nil), forCellReuseIdentifier: Constants.TableCellsIdentifier.favorites)
-        
-        tableView.delegate = dataSource
-        tableView.dataSource = dataSource
-        dataSource.newsArray = newsArray
-        dataSource.cellIdentifier = Constants.TableCellsIdentifier.favorites
+        loadingView.initView(delegate: self)
+        setupTableView()
+        displaySavedArticlesOnScreen()
     }
     
-    
+    func setupTableView() {
+        tableView.register(UINib(nibName: Constants.NibNames.favorites, bundle: nil), forCellReuseIdentifier: Constants.TableCellsIdentifier.favorites)
+        self.dataSource = TableViewDataSourceManager(
+            models: newsArray,
+            reuseIdentifier: Constants.TableCellsIdentifier.favorites
+        ) { savedArticle, cell in
+            let currentCell = cell as! SavedArticleCell
+            currentCell.articleTitle.text = savedArticle.articleTitle
+            currentCell.articleTopic.setTitle(savedArticle.topic, for: .normal)
+        }
+        tableView.delegate = self
+        tableView.dataSource = self.dataSource
+        tableView.rowHeight = 115.0
+    }
     
     override func viewWillAppear(_ animated: Bool) {
         super.viewWillAppear(animated)
         navigationController?.isNavigationBarHidden = true
     }
-
+    
     func viewWillDisppear(_ animated: Bool) {
         super.viewWillDisappear(animated)
         navigationController?.isNavigationBarHidden = false
     }
+    
+    
+    func displaySavedArticlesOnScreen() {
+        
+        DispatchQueue.main.async {
+            self.loadingView.isHidden = false
+            self.loadingView.loadIndicator.startAnimating()
+        }
+        
+        self.fetchNewsFromAPI() {
+            
+            DispatchQueue.main.async {
+                self.loadingView.loadIndicator.stopAnimating()
+                self.loadingView.isHidden = true
+            }
+            
+            if self.newsArray.count == 0 {
+                //TO DO: tell user there are no saved articles
+            }
+        }
+    }
+    
+    
+    func fetchNewsFromAPI(completionHandler: @escaping () -> ()) {
```

Comment:
Lets put all the network functions and non UI related functions in ArticleRepository class.
Lets talk how to do it also tomorow.

Reply:
Fixed in the MVVM pull request
@abenartz